src/common/src/hash/consistent_hash/mapping.rs

```diff
@@ -139,10 +139,10 @@ impl<T: VnodeMappingItem> VnodeMapping<T> {
         }
     }
 
-    /// Create a vnode mapping where all vnodes are mapped to the same single item.
+    /// Create a vnode mapping with the single item. Should only be used for singletons.
+    // TODO(var-vnode): make vnode count 1, also `Distribution::vnode_count`
     pub fn new_single(item: T::Item) -> Self {
-        // TODO(var-vnode): always 1 correct?
-        Self::new_uniform(std::iter::once(item), 1)
+        Self::new_uniform(std::iter::once(item), VirtualNode::COUNT)
```

Comment:
Should this vnode count (`VirtualNode::COUNT`) be passed from caller as an argument?

Reply:
Can refer to https://github.com/risingwavelabs/risingwave/pull/18515/files#diff-776f2681803414a56f539f2c98cc63a749bad4d8531d3d3191fc5df9c7efc004R61, where I find keeping using `256` for the vnode count of singleton fragments simplifies backwards compatibility.